site/src/Librecores/ProjectRepoBundle/Controller/ProjectController.php

```diff
@@ -561,4 +620,40 @@ private function makeGraph($languages, $multiDim = true)
 
         return $graph;
     }
+
+    /**
+     * Validate a Project Classification
+     *
+     * @param string $classification holds a clssification for a project
+     *
+     * @return bool
+     */
+    private function classificationValidation($classification)
+    {
+        $em = $this->getDoctrine()->getManager()
+            ->getRepository(ClassificationHierarchy::class);
+
+        $classificationHierarchy = $em->findByParent(null);
+        $classifications = [];
+        foreach ($classificationHierarchy as $classHyc) {
+            $classifications[$classHyc->getName()] = $classHyc;
+        }
+        $categories = explode('::', $classification);
+        $i = 0;
+        $count = count($categories);
+        foreach ($categories as $category) {
+            if (array_key_exists($category, $classifications)) {
+                $classificationHierarchy = $em->findByParent($classifications[$category]);
+                $classifications = [];
+                foreach ($classificationHierarchy as $classHyc) {
+                    $classifications[$classHyc->getName()] = $classHyc;
+                }
+                if (++$i == $count) {
+                    return true;
+                }
```

Comment:
You don't need the count here. It should be sufficient to return true after the end of the foreach loop (and leave the "return false") where it is currently. Even more understandable code could be created by returning early, something like that:

```php
foreach ($categories as $category) {
    if (!array_key_exists($category, $classifications)) {
        return false;
    }

    // now load subcategories
    // ...
}

return true;
```